src/Temporalio/Workflows/ChildWorkflowSignalOptions.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Threading;
+
+namespace Temporalio.Workflows
+{
+    /// <summary>
+    /// Options for child workflow signalling.
+    /// </summary>
+    public class ChildWorkflowSignalOptions : ICloneable
+    {
+        /// <summary>
+        /// Gets or sets the cancellation token to cancel the signal request. If the signal is
+        /// already sent, this does nothing. If unset, this defaults to the workflow cancellation
+        /// token.
+        /// </summary>
+        public CancellationToken? CancellationToken { get; set; }
```

Comment:
@bergundy and @Sushisource - I need help knowing whether this (and the one for external) has any user-facing value. I am trying to understand the value of `CancelSignalWorkflow` in core as it relates to users.

Reply:
Has _any_? Sure, I suppose there's some in theory. In practice it seems crazy unlikely. I would say you can leave it out.

Reply:
I guess the only reason why you'd want this is to be able to resolve the task returned to the user in case the token is cancelled internally, which is a good enough reason to keep this IMHO.
It makes the cancellation experience for the SDK's async APIs consistent.